### PR TITLE
Plugin statuses

### LINKED ISSUE #287: Report GVKs for used plugins in status
It is impossible to find out group/version/kind of an object produced by a plugin, used in a Bundle by just looking at it. So object's GVK should be reported for each used plugin in the Bundle status.

### PR DESCRIPTION
**Pull request by ash2k**
Fixes #287.

### CODE REVIEW

File: pkg/controller/bundlec_test/zz_plumbing_for_test.go [hunk 1]
```diff
@@ -327,6 +328,47 @@ func (tc *testCase) verifyPlugins(t *testing.T) {
...
+	}
+	pluginNames := map[smith_v1.PluginName]struct{}{}
+	for _, res := range tc.bundle.Spec.Resources {
+		if res.Spec.Plugin == nil {
+			continue
+		}
+		pluginNames[res.Spec.Plugin.Name] = struct{}{}
+	}
+	assert.Len(t, bundle.Status.PluginStatuses, len(pluginNames))
+nextPlugin:
```
Comment (nilebox) [inline pkg/controller/bundlec_test/zz_plumbing_for_test.go]: not a fan of goto operator, but for tests fine.

File: pkg/controller/bundlec/bundle_sync_task.go [hunk 2]
```diff
@@ -414,6 +414,41 @@ func (st *bundleSyncTask) handleProcessResult(retriable bool, processErr error)
...
+			}
+			pluginName := res.Spec.Plugin.Name
+			if _, ok := name2status[pluginName]; ok {
+				continue // Already reported
+			}
+			name2status[pluginName] = struct{}{}
+			var pluginStatus smith_v1.PluginStatus
+			pluginContainer, ok := st.pluginContainers[pluginName]
+			if ok {
+				describe := pluginContainer.Plugin.Describe()
```
Comment (nilebox) [inline pkg/controller/bundlec/bundle_sync_task.go]: What if plugin returns `nil`?
Probably fine to explode in that case, rather than continue running Smith with invalid plugin.

File: WORKSPACE [hunk 3]
```diff
@@ -3,19 +3,20 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 http_archive(
     name = "io_bazel_rules_go",
     sha256 = "c1f52b8789218bb1542ed362c4f7de7052abcf254d865d96fb7ba6d44bc15ee3",
-    url = "https://github.com/bazelbuild/rules_go/releases/download/0.12.0/rules_go-0.12.0.tar.gz",
+    urls = ["https://github.com/bazelbuild/rules_go/releases/download/0.12.0/rules_go-0.12.0.tar.gz"],
```
Comment (nilebox) [inline WORKSPACE]: Where is this change coming from? New version of Bazel? Seems irrelevant to PR.

File: pkg/apis/smith/v1/types.go [hunk 4]
```diff
@@ -149,12 +156,21 @@ func (bc *BundleCondition) String() string {
 	return buf.String()
 }
 
+type PluginStatus struct {
+	Name    PluginName      `json:"name"`
+	Group   string          `json:"group"`
```
Comment (nilebox) [inline pkg/apis/smith/v1/types.go]: use `metav1.GroupVersionKind` there (or its copy if you don't want to reuse `metav1` types in API)?

File: pkg/controller/bundlec/bundle_sync_task.go (hunk 2 above)
Comment (ash2k) [reply pkg/controller/bundlec/bundle_sync_task.go]: Hm, I wonder why it is a pointer in the first place? Should probably change it to a non-pointer.
We call this function in a lot of places and never assume it can return a `nil` pointer.

File: pkg/controller/bundlec_test/zz_plumbing_for_test.go (hunk 1 above)
Comment (ash2k) [reply pkg/controller/bundlec_test/zz_plumbing_for_test.go]: I think this is better than an inline anon function and a `return`.

File: WORKSPACE (hunk 3 above)
Comment (ash2k) [reply WORKSPACE]: This is in another commit, just a cleanup.

File: pkg/apis/smith/v1/types.go (hunk 4 above)
Comment (ash2k) [reply pkg/apis/smith/v1/types.go]: That is what I did initially. But I didn't want to create a sub object for these fields just to re-use the struct.

File: pkg/controller/bundlec/bundle_sync_task.go (hunk 2 above)
Comment (ash2k) [reply pkg/controller/bundlec/bundle_sync_task.go]: I'll send a follow up to remove the pointer here.

Comment (ash2k) [reply pkg/controller/bundlec/bundle_sync_task.go]: Or maybe not. Not sure about this one....
